report_qweb_signer/models/report.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+# © 2015 Antiun Ingenieria S.L. - Antonio Espinosa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import os
+import subprocess
+import tempfile
+import base64
+import time
+from contextlib import closing
+
+from openerp import SUPERUSER_ID, models, api, _
+from openerp.exceptions import Warning, AccessError
+from openerp.tools.safe_eval import safe_eval as eval
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class Report(models.Model):
+    _inherit = 'report'
+
+    def _signer_bin(self, opts):
+        return '/usr/local/bin/PortableSigner -n ' + opts
+
+    def _user_company_id_get(self, cr, uid, context=None):
+        m_users = self.pool.get('res.users')
+        user = m_users.browse(cr, SUPERUSER_ID, uid, context=context)
+        return user.company_id.id
+
+    def _certificate_get(self, cr, uid, ids, report, context=None):
+        if report.report_type != 'qweb-pdf':
+            _logger.info(
+                "Can only sign qweb-pdf reports, this one is '%s' type" %
+                report.report_type)
+            return False
+        m_cert = self.pool.get('report.certificate')
+        company_id = self._user_company_id_get(cr, uid, context=context)
+        certificate_ids = m_cert.search(cr, uid, [
+            ('company_id', '=', company_id),
+            ('model_id', '=', report.model)], context=context)
+        if not certificate_ids:
+            _logger.info(
+                "No PDF certificate found for report '%s'" %
+                report.report_name)
+            return False
+        for cert_id in certificate_ids:
+            cert = m_cert.browse(cr, uid, cert_id, context=context)
+            # Check allow only one document
+            if cert.allow_only_one and len(ids) > 1:
+                _logger.info(
+                    "Certificate '%s' allows only one document, "
+                    "but printing %d documents" %
+                    (cert.name, len(ids)))
+                continue
+            # Check domain
+            if cert.domain:
+                m_model = self.pool[cert.model_id.model]
+                domain = [('id', 'in', tuple(ids))]
+                domain = domain + eval(cert.domain)
+                doc_ids = m_model.search(cr, uid, domain, context=context)
+                if not doc_ids:
+                    _logger.info(
+                        "Certificate '%s' domain not satisfied" % cert.name)
```

Comment:
Here logging message is because domain configured is not returning any result. This means that this PDF must not be signed.
Domain is provided but not satisfied its condition.